### PR TITLE
Adjust the IconButton constraint to match its size

### LINKED ISSUE #5763: Big IconButtons are clipped
```
import 'package:flutter/material.dart';

class BigIconButton extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      appBar: new AppBar(title: new Text('Big IconButton')),
      body: new Center(
        // Does what you'd expect:
        // child: new Icon(Icons.android, size: 128.0)
        // Does not:
        child: new IconButton(
          icon: new Icon(Icons.android, size: 128.0),
          // reduntantly specifying 'size: 128.0' doesn't help 
          onPressed: () {}
        )
      )
    );
  }
}

void main() => runApp(new MaterialApp(home: new BigIconButton()));
```
screenshot1

### PR DESCRIPTION
**Pull request by jason-simmons**
Fixes https://github.com/flutter/flutter/issues/5763

### CODE REVIEW

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/material/material.dart [hunk 1]
```diff
@@ -321,11 +321,14 @@ class _RenderInkFeatures extends RenderProxyBox implements MaterialInkController
...
     RectCallback rectCallback,
-    VoidCallback onRemoved
+    VoidCallback onRemoved,
+    double radiusOverride,
   }) {
     double radius;
     RectCallback clipCallback;
-    if (containedInkWell) {
+    if (radiusOverride != null) {
+      radius = radiusOverride;
```
Comment (jason-simmons) [inline packages/flutter/lib/src/material/material.dart]: done

**Comment by Hixie** [comment]
Hm, actually, this changes the semantics for some cases.

File: packages/flutter/lib/src/material/ink_well.dart [hunk 2]
```diff
@@ -36,7 +36,8 @@ class InkResponse extends StatefulWidget {
     this.onLongPress,
     this.onHighlightChanged,
     this.containedInkWell: false,
-    this.highlightShape: BoxShape.circle
+    this.highlightShape: BoxShape.circle,
+    this.radius: InkSplash.defaultRadius,
```
Comment (Hixie) [inline packages/flutter/lib/src/material/ink_well.dart]: I guess we should default this to null.

Comment (jason-simmons) [inline packages/flutter/lib/src/material/ink_well.dart]: done

**Pull request merged**